pynamodb_attributes/unicode_delimited_tuple.py

```diff
@@ -68,3 +72,20 @@ def serialize(self, value: T) -> str:
                 f"Tuple elements may not contain delimiter '{self.delimiter}'",
             )
         return self.delimiter.join(strings)
+
+    def _parse_value(self, str_value: str, type_: Type[Any]) -> Any:
+        if hasattr(type_, "__args__"):
```

Comment:
Are you trying to parse Union types here?

Reply:
Yes. This works for both, 3.8 and 3.9. In 3.9 `Optional[T]` types are used as `Union[NoneType, T]`, but `__args__` has the specific types.

Reply:
I think `__args__` exist in any generic type, not just `Union`, so at the very least I'd ensure the type is `Union`. But honestly I'd omit this. The user would get an `TypeError: Cannot instantiate typing.Union` error and that'll be enough to understand that the NamedTuple should consist of simple types.

Reply:
So leave as is?
In this case:
```
>>> typing.Union[int, str].__args__
(<class 'int'>, <class 'str'>)
```

Reply:
Yeah, don't try to parse `__args__`. Simply `t(value)` and be done with it (like we did before).

Reply:
Sounds good, I think It would be _nicer_ to do so, I'll go ahead and remove this.

Reply:
Constructs like `typing.Union` within the Python ecosystem are intended for type checkers to parse, not runtime. I doubt there's many counter-examples, perhaps in specialized libraries like pydantic? A DB ORM making any sense of them is probably an example of a library doing things you don't expect it to :)

One possibility could be to even do away with the option of parsing into an `int`. In that case, you can forego all this annotations inspection and define:
```
T = TypeVar("T", bound=Tuple[str, ...])
```
and only do `return self.tuple_type(*values)` in the deserializer.

However, this runs counter to the sample code. :)

Reply:
@ikonst Removed!